data_model/src/transaction.rs

```diff
@@ -165,12 +166,27 @@ pub mod model {
         pub signatures: SignaturesOf<TransactionPayload>,
     }
 
-    /// Transaction rejected by the validator
+    /// Transaction Value used in Instructions and Queries
     #[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
     #[ffi_type]
-    pub struct RejectedTransaction {
-        pub transaction: VersionedSignedTransaction,
-        pub error: error::TransactionRejectionReason,
+    pub struct TransactionValue {
+        /// Committed transaction
+        pub tx: VersionedSignedTransaction,
+        /// Reason of rejection
+        pub error: Option<error::TransactionRejectionReason>,
+    }
+
+    /// `TransactionQueryResult` is used in `FindAllTransactions` query
+    #[derive(
+        Debug, Clone, PartialEq, Eq, Getters, Decode, Encode, Deserialize, Serialize, IntoSchema,
+    )]
+    #[getset(get = "pub")]
+    #[ffi_type]
+    pub struct TransactionQueryResult {
```

Comment:
please return this struct in the `query.rs` module. It only exists because of querying